File: refactoring-practice-1/src/main/java/dojo/extract_subclass/JobItem.java
package dojo.extract_subclass;

/*
 * A labor mezőtől függően az osztály máshogyan számol. Emeljük ki ezt a viselkedést egy LaborItem osztályba!
 */
public class JobItem {

	private final double unitPrice;
	private final int quantity;
	private final Employee employee;
	private final boolean labor;
	
	public JobItem(double unitPrice, int quantity, Employee employee, boolean labor) {
		super();
		this.unitPrice = unitPrice;
		this.quantity = quantity;
		this.employee = employee;
		this.labor = labor;
	}
	
	public double getTotalPrice() {
		return getUnitPrice() * quantity;
	}
	
	public double getUnitPrice() {
		return (labor ? employee.getRate() : unitPrice);
	}
	
	public int getQuantity() {
		return quantity;
	}
	
	public Employee getEmployee() {
		return employee;
	}
	
}

File: refactoring-practice-1/src/main/typescript/extract_subclass/job-item.ts
/*
 * A labor mezőtől függően az osztály máshogyan számol. Emeljük ki ezt a viselkedést egy LaborItem osztályba!
 */
import {Employee} from "./employee.js";

export class JobItem {

    public constructor(private unitPrice: number, private quantity: number, private employee: Employee, private labor: boolean) {
    }

    public getTotalPrice() {
        return this.getUnitPrice() * this.quantity;
    }

    public getUnitPrice() {
        return (this.labor ? this.employee.getRate() : this.unitPrice);
    }

    public getQuantity() {
        return this.quantity;
    }

    public getEmployee() {
        return this.employee;
    }

}
